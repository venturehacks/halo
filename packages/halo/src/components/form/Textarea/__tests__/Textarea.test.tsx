import { render } from 'enzyme';
import React from 'react';

import { Textarea } from '../index';

describe('Textarea', () => {
  test('smoke', () => {
    const component = render(
      <Textarea onChange={() => {}} value="textarea value" />,
    );

    expect(component.text()).toContain('textarea value');
  });

  describe('snapshot', () => {
    describe('size', () => {
      test('sm', () => {
        const component = render(
          <Textarea onChange={() => {}} size="sm" value="small" />,
        );
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(
          <Textarea onChange={() => {}} value="medium" />,
        );
        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = render(
          <Textarea onChange={() => {}} size="lg" value="large" />,
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
});
