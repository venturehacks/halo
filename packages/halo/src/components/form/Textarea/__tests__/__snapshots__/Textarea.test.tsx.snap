// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea snapshot size lg 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a ring-inset border rounded focus:ring-dark-link bg-white  focus:ring-1 p-3  h-12 border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-dark-aa w-auto"
>
  large
</textarea>
`;

exports[`Textarea snapshot size md 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a ring-inset border rounded focus:ring-dark-link bg-white  focus:ring-1 px-3 py-2 h-10 border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-dark-aa w-auto"
>
  medium
</textarea>
`;

exports[`Textarea snapshot size sm 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a ring-inset border rounded focus:ring-dark-link bg-white  focus:ring-1 px-3 py-1 h-8 border-gray-500 disabled:text-dark-a disabled:bg-gray-200 disabled:border-dark-aa w-auto"
>
  small
</textarea>
`;
